### DISCUSSION TITLE
email is not returned by signin callback or session

### DISCUSSION THREAD

**Discussion by ats1999** [nextauthjs/next-auth#2468] [Help]
### Question 💬

I used signIn callback from here [docs](https://next-auth.js.org/configuration/callbacks#session-callback)

I tried to log everything,
```js
async signIn(user, account, profile) {
      console.log("Singin....", { user,account,profile})
      return true
    }
```
What i got
```js
user: {
    id: ...,
    name: 'Rahul kumar',
    email: null,
    image: 'https://avatars.githubusercontent.com/u/54087826?v=4'
  },
  account: {
    provider: 'github',
    type: 'oauth',
    ...
  },
  profile: {
    login: 'ats1999',
    ...
  }
```

### How to reproduce ☕️

Use the signin callback and log everything.

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by balazsorban44** [comment]
You might be missing a required scope for GitHub to return the email?

**Reply by ats1999** [reply]
I did't found any documention for scope.

**Reply by balazsorban44** [reply]
https://next-auth.js.org/providers/github#options

https://next-auth.js.org/configuration/providers#options

https://github.com/nextauthjs/next-auth/blob/main/src/providers/github.js

**Comment by verekia** [comment]
See this issue: https://github.com/nextauthjs/next-auth/issues/374

**Reply by balazsorban44** [reply]
https://github.com/nextauthjs/next-auth/issues/374#issuecomment-931741760
